x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/AbstractSqlQueryRequest.java

```diff
@@ -233,6 +234,11 @@ public ActionRequestValidationException validate() {
         return validationException;
     }
 
+    protected boolean isClientCompatible() {
```

Comment:
Do you think this method better belongs to `SqlVersion` class where `hasVersionCompatibility()` is also defined?

Reply:
Yeh, I had a hesitation initially, but it does belong there more naturally. Thanks for raising it.